src/bridge-canister/src/operation_store.rs

```diff
@@ -240,6 +240,14 @@ where
             .collect()
     }
 
+    /// Retrieve all memos for a given user_id in the store.
+    pub fn get_memos_by_user(&self, user_id: &H160) -> Vec<Memo> {
+        self.memo_operation_map
```

Comment:
This forces us to loop through all available memos, instead, we would like to invert the map structure from `memo_operation_map: StableMultimap<Memo, H160, OperationId, M>,` to `memo_operation_map: StableMultimap<H160, Memo, OperationId, M>,`